### DISCUSSION TITLE
Exception during Telegraph::store for large videos

### DISCUSSION THREAD

**Discussion by YAlex84** [def-studio/telegraph#607] [Q&A] [answered]
I'm trying to save a video while a webhook is triggered. Small files, up to about 10MB, are saved without problems. Larger files are not saved and the exception "Failed to retreive info for file" is thrown during Telegraph::store. What could be the problem?

`Telegraph::store($this->message->video(), storage_path() . '/temp');`

$this->message->video() dumps for small and large videos look the same good.

**Comment by fabio-ivona** [comment]
Hi @YAlex84 it seems that the issue comes from telegram's side

could you please dump `$response->dump()` in vendor/defstudio/telegraph/Concerns/StoresFiles.php::32 ?

**Reply by YAlex84** [reply]
Hi @fabio-ivona thanks for the answer. The dump is as follows:

`{"ok":false,"error_code":400,"description":"Bad Request: file is too big"}`

Actually, the problem is indicated in the description) Another question is, what to do about it?

**Reply by fabio-ivona** [reply, marked as answer]
What is the size of your file? Telegram has a 2gb max size for non-premium users

Edit: it seems that Bots have a lower file size limit for downloads: 20MB:

https://core.telegram.org/bots/api#getfile

I can't see a workaround for this, sorry

**Reply by YAlex84** [reply]
It looks like there really is no solution... But it’s good that the reason and the download limit are now known. Thank you for participating!

**Reply by fabio-ivona** [reply]
We have updated our docs to include this limitation, thanks for pointing this out!
